pixl_core/src/core/project_config/secrets.py

```diff
@@ -80,18 +93,31 @@ def _fetch_secret(secret_name: str) -> str:
     This helper is intentionally defined outside the Azure Keyvault to prevent memory leaks (see
     ruff rule B019).
 
+    :param client: the Azure Key Vault client
     :param secret_name: the name of the secret to fetch
     :return: the requested secret's value
     """
-    kv_name = config("AZURE_KEY_VAULT_NAME")
-    key_vault_uri = f"https://{kv_name}.vault.azure.net"
-
-    credentials = DefaultAzureCredential()
-    client = SecretClient(vault_url=key_vault_uri, credential=credentials)
-
     secret = client.get_secret(secret_name).value
 
     if secret is None:
         msg = f"Azure Key Vault secret {secret_name} is None"
         raise ValueError(msg)
     return str(secret)
+
+
+@lru_cache
+def _create_secret(client: SecretClient, secret_name: str, secret_value: str) -> None:
```

Comment:
all arguments need to be hashable for caching to be enabled. So you could make SecretClient hashable or keep this as before where it was being cached